web/src/app/components/Lobby.tsx

```diff
@@ -1,19 +1,21 @@
-import React, { useRef, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
```

Comment:
So for the logic of finding out who is host is done in this Lobby component but it easily could have been done in the `Game.tsx` component as well, and we could have just passed a boolean prop into this `Lobby` component. I'm not sure which one is better so what are your thoughts?

Reply:
I think putting it in Game.tsx is better. instead of having this useEffect, we can just put the check for host inside `socket.on('allPlayers')`. That way we can also be sure the host check is only run when a player joins/leaves (addressing the useEffect problem you mentioned about in the other comment)

something like:

```
        socket.on('allPlayers', (players) => {
            setPlayers(players);
            setIsCurrPlayerHost(players[currPlayerID].host);
        })
```

Reply:
Yeah I agree, but I think I have to useEffect() here because of react's state changes. `currPlayerID` is always empty so I can't do anything in that socketevent

Reply:
currPlayerId should be set on the connect event. I think this will work

Reply:
I tried and it didn't work.

Reply:
you're right, since currPlayerID uses useState, it's not updated until the next render. We can just use `useRef `instead.

useRef is similar to state except it doesn't trigger a re-render when the value is changed. But since currPlayerID just used internally and not for any UI, useRef is better anyway